src/Type/BitwiseFlagAnalyser.php

```diff
@@ -0,0 +1,99 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\BinaryOp\BitwiseOr;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Name\FullyQualified;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\Constant\ConstantIntegerType;
+
+final class BitwiseFlagAnalyser
+{
+
+	public function __construct(private ReflectionProvider $reflectionProvider)
+	{
+	}
+
+	public function exprContainsConstant(Expr $expr, Scope $scope, string $constName): TrinaryLogic
```

Comment:
Something should indicate here this is about BitwiseOr and nothing else.

Reply:
renamed it to `bitwiseOrContainsConstant`